.github/build_real.sh

```diff
@@ -7,4 +7,5 @@ if [[ $IMAGE == "private" ]]; then
     cp -r $HOME/SNOPT/* pyoptsparse/pySNOPT/source
 fi
 
-pip install .[optview,testing]
+pip install --upgrade numpy
```

Comment:
This actually related to your comment:
> when I try to import any of the compiled libraries. Looking through the log, it seems that due to the presence of pyproject.toml, everything is built from a temporary directory so the f2py/numpy includes point to locations that no longer exist. I'm not sure what exactly is going on, but if I remove the pyproject.toml file then everything works fine for me. Are you seeing the same thing on your end @jackm97?

It seems `pip install .` doesn't check the numpy version of the environment it's installing to. It just installs the latest `numpy` version for the build. So once the package is installed and used it fails because it was built with a different version of `numpy` than what is installed in the target environment

Reply:
https://numpy.org/devdocs/user/depending_on_numpy.html#build-time-dependency

Reply:
In that case it seems we should use `==1.16` instead of `>=1.16` for numpy?

Reply:
It seems the meson build doesn't like that...

Reply:
or the tests on my local machine

Reply:
I think I managed to fix it with [10ba333](https://github.com/mdolab/pyoptsparse/pull/300/commits/10ba3330066767b53b70d2364d0659e5a01c181c)